neurodamus/core/stimuli.py

```diff
@@ -201,22 +217,17 @@ def add_shot_noise(self, tau_D, tau_R, rate, amp_mean, amp_var, duration, dt=0.2
 
         exp_scale = 1 / rate  # scale parameter of exponential distribution of time intervals
         rng.negexp(exp_scale)
-        iei = Neuron.h.Vector(napprox)
+        iei = Nd.h.Vector(napprox)
         iei.setrand(rng)  # generate inter-event intervals
 
-        ev = Neuron.h.Vector()
+        ev = Nd.h.Vector()
         ev.integral(iei, 1).mul(1000)  # generate events in ms
         # add events if last event falls short of duration
-        while ev[-1] < duration:
-            iei_new = Neuron.h.Vector(100)  # generate 100 new inter-event intervals
-            iei_new.setrand(rng)  # here rng is still negexp
-            ev_new = Neuron.h.Vector()
-            ev_new.integral(iei_new, 1).mul(1000).add(ev[-1])  # generate new shifted events in ms
-            ev.append(ev_new)  # append new events
+
```

Comment:
Is the last event no more added? The comment is still there.

Reply:
ah thank you. that is a typo